apps/i3wm/i3wm.py

```diff
@@ -29,6 +63,44 @@ def i3wm_shell():
         key = settings.get("user.i3_mod_key")
         actions.key(f"{key}-enter")
 
+    def i3wm_focus(what: str):
+        """Move focus"""
+        subprocess.check_call(("i3-msg", "focus", what))
+
+    def i3wm_switch_to_workspace(which: Union[str, int]):
+        """Focus the specified workspace"""
+        if isinstance(which, int):
+            subprocess.check_call(("i3-msg", "workspace", "number", str(which)))
+        else:
+            subprocess.check_call(("i3-msg", "workspace", which))
+
+    def i3wm_show_scratchpad():
+        """Focus/cycle/hide the scratchpad"""
+        subprocess.check_call(("i3-msg", "scratchpad", "show"))
+
+    def i3wm_move(to: str):
+        """Move the focused container"""
+        subprocess.check_call(("i3-msg", "move", to))
+
+    def i3wm_move_to_workspace(which: Union[str, int]):
+        """Move the focused container to the specified workspace"""
+        if isinstance(which, int):
+            subprocess.check_call(
+                ("i3-msg", "move", "container", "to", "workspace", "number", str(which))
```

Comment:
If we decide to override `user.desktop`, we should also override `user.window_move_desktop(number: int)`, and `window_move_desktop_{left,right}` too.